surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -48,15 +54,28 @@ private boolean isUsageSet() {
         for (final String strategy : strategies) {
 
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
-            final List<? extends Class<?>> tests = plannerForStrategy.getTests().stream().map(testClass -> {
-                try {
-                    return Class.forName(testClass);
-                } catch (ClassNotFoundException e) {
-                    throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
-                }
-            }).collect(Collectors.toList());
+            final List<? extends Class<?>> tests = plannerForStrategy.getTests()
+                .stream().filter(this::validateClassPathAndTestClass)
+                .map(testClass -> {
+                    try {
+                        return Class.forName(testClass);
+                    } catch (ClassNotFoundException e) {
+                        throw new IllegalStateException("Failed while obtaining strategy for " + strategy, e);
+                    }
+                }).collect(Collectors.toList());
             orderedTests.addAll(tests);
         }
         return orderedTests;
     }
+
+    private boolean validateClassPathAndTestClass(String testClass) {
+        Class<?> aClass;
+        try {
+            aClass = testClassLoader.loadClass(testClass);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+
+        return jUnit4TestChecker.accept(aClass);
```

Comment:
I'm not sure we should really care about it. Notice that for surefire it doesn't matter too. It just provides list of files and the rest is taken care by underlying test runner. It will validates and reports if the class has a structure according to the framework of choice or not.